File: tests/testdata/test_fas_code_reader/sample.cpp
#include <iostream> 
#include <string>    
#include "my_utility.h" 

class Animal {
public:
    void speak() {
        std::cout << "Animal sound" << std::endl;
    }
};

int helper() {
    std::cout << "I'm a helper" << std::endl;
}

int main() {
    for (int i = 1; i <= 5; i++) {
        for(int j = 2; j <= 10; j++){
            printf("%d\n", i);
            printf("%d\n", j);
        }
    }

    while(int x < 5){
        x = x + 1;
    }

    return 0;
}

File: tests/testdata/test_fas_code_reader/sample.rs
// Sample Rust file for testing library extraction
use std::collections::HashMap;
use std::io::Result;
use tokio::runtime::Runtime;
use crate::internal::module;
use self::local_function;
use super::parent_module;
use super::super::grandparent;

fn helper() {
    println!("I'm a helper");
}

struct Animal {
    name: String,
}

impl Animal {
    fn speak(&self) {
        println!("Animal sound");
    }
}

fn main() {
    for i in 1..6 {
        for j in 1..6 {
            println!("i is: {}", i);
            println!("j is: {}", j);
        }
    }
}